R/panel.r

```diff
@@ -49,6 +49,7 @@ train_layout <- function(panel, facet, data, plot_data) {
 map_layout <- function(panel, facet, data, plot_data) {
   lapply(data, function(data) {
     if (is.waive(data)) data <- plot_data
+    if (is.function(data)) data <- fortify(data(plot_data))
```

Comment:
This doesn't seems like the right place - `compute_aesthetics()` would be better